src/NumericalAlgorithms/Interpolation/PredictedZeroCrossing.cpp

```diff
@@ -16,4 +16,15 @@ double predicted_zero_crossing_value(const std::vector<double>& x_values,
   return -coefficients[0]/coefficients[1];
 }
 
+std::vector<double> predicted_zero_crossing_value(
```

Comment:
Above in the case for doubles, add a check that x_values and y_values have the same size?